### PR TITLE
44_arkouda_metrics_exporter

### PR DESCRIPTION
**Pull request by hokiegeek2**
This PR corresponds to the arkouda_metrics_exporter project that enables the export of Arkouda metrics to Prometheus by way of a prometheus_client-based approach

### CODE REVIEW

**Review by Ethan-DeBandi99 — CHANGES_REQUESTED** [review]
I have not worked with prometheus previously, but on the surface after some quick research, this looks good.

There are a few house keeping things (mostly to keep things configured properly). At the top level, we have typically been requiring `test` folders in addition to `client` even when empty (though I'd argue we should always have testing). I wanted to put this up and I am going to do some more digging on prometheus for my own knowledge.

File: arkouda_metrics_exporter/client/arkouda_metrics_exporter/metrics.py [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+
+    def _updateServerMetrics(self, metric: Metric) -> None:
+        self.numberOfConnections.labels(arkouda_server_name=self.arkoudaMetricsServerName).set(
+            metric.value
+        )
+
+    def _updateReportedTimestamp(self, value: datetime) -> None:
+        self.reportedTimestamp.labels(arkouda_server_name=self.arkoudaMetricsServerName).set(
+            value.timestamp() * 1000
+        )
```
Comment (pierce314159) [inline arkouda_metrics_exporter/client/arkouda_metrics_exporter/metrics.py:483]: I might be missing it but i don't see where this update method is used

**Review by pierce314159 — COMMENTED** [review]
I don't have much experience with this but the code looks good to me. I do have a question about one of the update methods

File: arkouda_metrics_exporter/client/arkouda_metrics_exporter/metrics.py (hunk 1 above)
Comment (hokiegeek2) [reply arkouda_metrics_exporter/client/arkouda_metrics_exporter/metrics.py:483]: @pierce314159 good catch! Yeah, that method is no longer needed, will remove

Comment (hokiegeek2) [reply arkouda_metrics_exporter/client/arkouda_metrics_exporter/metrics.py]: @pierce314159 just removed it and retested.

**Review by pierce314159 — APPROVED** [review]

**Review by Ethan-DeBandi99 — APPROVED** [review]
This looks good to me

File: arkouda_metrics_exporter/client/arkouda_metrics_exporter/metrics.py (hunk 1 above)
Comment (Ethan-DeBandi99) [reply arkouda_metrics_exporter/client/arkouda_metrics_exporter/metrics.py]: Resolving this because I verified it was removed.

**Pull request merged**